solidity/dashboard/src/pages/rewards/TBTCRewardsPage.jsx

```diff
@@ -0,0 +1,187 @@
+import EmptyStatePage from "./EmptyStatePage"
+// import { colors } from "../../constants/colors"
+
+const TBTCRewardsPage = () => {
+  const dispatch = useDispatch()
+  const yourAddress = useWeb3Address()
+  const currentIntervalEndOf = ECDSARewardsHelper.intervalEndOf(
+    ECDSARewardsHelper.currentInterval
+  ).unix()
+
+  const {
+    ecdsaAvailableRewardsFetching,
+    ecdsaAvailableRewardsBalance,
+    ecdsaAvailableRewards,
+    ecdsaRewardsHistory,
+  } = useSelector((state) => state.rewards)
+
+  useEffect(() => {
+    dispatch({
+      type: "rewards/ecdsa_fetch_rewards_data_request",
+      payload: yourAddress,
+    })
+  }, [dispatch, yourAddress])
+
+  const withdrawRewards = useCallback(
+    (submitButtonPromise) => {
+      dispatch({
+        type: "rewards/ecdsa_withdraw",
+        payload: ecdsaAvailableRewards,
+        meta: submitButtonPromise,
+      })
+    },
+    [dispatch, ecdsaAvailableRewards]
+  )
+
+  return (
+    <>
+      <RewardsOverview
+        balance={ecdsaAvailableRewardsBalance}
+        isBalanceFetching={ecdsaAvailableRewardsFetching}
+        withdrawRewards={withdrawRewards}
+      />
+      <section className="tile rewards-countdown">
+        <h2 className="h2--alt">
+          Next rewards release:&nbsp;
+          <Timer targetInUnix={currentIntervalEndOf} />
+        </h2>
+      </section>
+      <LoadingOverlay
+        isFetching={ecdsaAvailableRewardsFetching}
+        skeletonComponent={<DataTableSkeleton columns={4} />}
+      >
+        <section className="tile rewards-history">
+          <TBTCRewardsDataTable data={ecdsaRewardsHistory} />
+        </section>
+      </LoadingOverlay>
+    </>
+  )
+}
+
+const RewardsOverview = ({ balance, isBalanceFetching, withdrawRewards }) => {
+  return (
+    <section className="tile rewards__overview--tbtc">
+      <div className="rewards__overview__balance">
+        <h2 className="h2--alt text-grey-70 mb-1">tBTC Rewards</h2>
+        {isBalanceFetching ? (
+          <TokenAmountSkeleton />
+        ) : (
+          <TokenAmount amount={balance} currencySymbol="KEEP" />
+        )}
+      </div>
+      <div className="rewards__overview__period">
+        <h5 className="text-grey-70">current rewards period</h5>
+        <span className="rewards-period__date">
+          {ECDSARewardsHelper.periodOf(ECDSARewardsHelper.currentInterval)}
+        </span>
+        <div className="rewards-period__remaining-periods">
+          <Icons.Time width="16" height="16" className="time-icon--grey-30" />
+          <span>84 rewards periods remaining&nbsp;</span>
+          <Tooltip simple delay={0} triggerComponent={Icons.MoreInfo}>
+            content
+          </Tooltip>
+        </div>
+      </div>
+      <div className="rewards__overview__withdraw">
+        <SubmitButton
+          className="btn btn-primary btn-lg"
+          onSubmitAction={withdrawRewards}
+          disabled={!gt(balance, 0)}
+        >
+          withdraw all
```

Comment:
I think that we could add some width to this button to make it looks more to the one in the [figma](https://www.figma.com/file/Su61MwHLrQn3ld24q85pMx/Token-Dashboard-dApp?node-id=7361%3A9712). Right now the button has less width than the text below which, in my opinion, doesn't look that good 🎨